src/cmd/src/frontend.rs

```diff
@@ -136,14 +130,16 @@ impl StartCommand {
             opts.logging.level = level;
         }
 
-        let tls_option = TlsOption::new(
+        let tls_opts = TlsOption::new(
             self.tls_mode.clone(),
             self.tls_cert_path.clone(),
             self.tls_key_path.clone(),
         );
 
         if let Some(addr) = self.http_addr.clone() {
-            opts.http_options.get_or_insert_with(Default::default).addr = addr;
+            if let Some(ref mut http_opts) = opts.http_options {
```

Comment:
style: Try this style, is more concise. Same as other places.
```suggestion
            if let Some(http_opts) = &mut opts.http_options {
```